perma_web/fabfile/dev.py

```diff
@@ -573,3 +573,89 @@ def check_s3_hashes():
                 m.update(buf)
         if m.hexdigest() != remote_paths[local_path]:
             print "Hash mismatch! Local: %s Remote: %s" % (m.hexdigest(), remote_paths[local_path])
+
+
+@task
+def check_storage():
+    """
+        Confirm that, for every link, there is a WARC in primary storage and in
+        secondary storage, and that their hashes match.
+
+        Derived from check_s3_hashes
+    """
+    from django.core.files.storage import default_storage
+    from perma.models import Link
+    from tqdm import tqdm
+
+    if not hasattr(default_storage, 'secondary_storage'):
+        print("No secondary storage to compare.")
+        return
+
+    counter = {
+        'match': 0,
+        'mismatch': 0,
+        'primary_only': 0,
+        'secondary_only': 0,
+        'no_warc': 0
+    }
+    cache_path = '/tmp/perma_path_list'
+
+    if not os.path.exists(cache_path):
+        print("Building state ...")
```

Comment:
Very optional readability suggestion, likely occurring to me only because I'm braindead: "building link list" and "using cached link list" might be clearer than "state". I also wonder if it's worth using .txt as an extension for cache_path, and calling it cache_file rather than cache_path. Feel free to ignore.